### ISSUE TITLE
Linux support

### ISSUE THREAD

**Issue by MostHated** [plule/leaprs#13]
Hey there, 
   I was wondering if there was any chance this might receive linux support? The actual Gemini framework has linux support: https://docs.ultraleap.com/linux/ I am guessing it must be newer? I only got my UL 3di device the other day, so I have not been keeping up until very recently. Just for giggles I attempted to change the build scripts LEAPSDK_LIB_PATH to the path of the libraries on linux, which were listed here:

```sh
The hand tracking client library, libLeapC.so, is installed to /usr/share/doc/ultraleap-hand-tracking-service with a symlink in /usr/lib/x86_64-linux-gnu/.
The corresponding header, LeapC.h, is installed to /usr/share/doc/ultraleap-hand-tracking-service/include, with a symlink in /usr/include.
```

Unfortunately, it keeps coming back with:

```rust
✗  LEAPSDK_LIB_PATH='/usr/share/doc/ultraleap-hand-tracking-service' cargo build                  
   Compiling leaprs v0.1.1-dev (/mnt/x/GitHub/plule/leaprs)
error: could not find native static library `LeapC`, perhaps an -L flag is missing?
```
I am assuming there must be more to it than that. What exactly, I unfortunately do not know.

Thanks,
-MH

**Comment by MostHated** [comment]
Awesome, that seems to have done the trick. It was able to recognize my camera and got to the "waiting for hand" message. Though, it did not detect one, but I believe I just need to figure out now to make sure it is aware of my setup. Running through the Web/Unity demos generated some json configs which contain orientation data and positioning values, since I am using the "above monitor" and "facing user" setup, it has the z-axis at -180 degrees and such. Not sure if that data is relevant with this?

<details>
<summary>TrackingConfig.json</summary>

```json
{
  "Mask": {
    "Lower": 0.0,
    "Upper": 0.0,
    "Right": 0.0,
    "Left": 0.0
  },
  "AllowImages": true,
  "CameraReversed": true,
  "AnalyticsEnabled": true

```

</details>


<details>
<summary>PhysicalConfig.json</summary>

```json
{
    "ScreenHeightM": 0.28821733593940737,
    "LeapPositionRelativeToScreenBottomM": {
        "x": -0.04785703495144844,
        "y": 0.378326952457428,
        "z": -0.037553731352090839
    },
    "LeapRotationD": {
        "x": -4.993865966796875,
        "y": 0.0,
        "z": 180.0
    },
    "ScreenRotationD": 0.0,
    "ScreenWidthPX": 2560,
    "ScreenHeightPX": 1080
}
```

</details>


<details>
<summary>InteractionConfig.json</summary>

```json
{
    "UseScrollingOrDragging": true,
    "UseSwipeInteraction": false,
    "DeadzoneRadius": 0.003000000026077032,
    "InteractionZoneEnabled": false,
    "InteractionMinDistanceCm": 0.0,
    "InteractionMaxDistanceCm": 25.0,
    "InteractionType": 2,
    "AirPush": {
        "SpeedMin": 150.0,
        "SpeedMax": 500.0,
        "DistAtSpeedMinMm": 42.0,
        "DistAtSpeedMaxMm": 8.0,
        "HorizontalDecayDistMm": 50.0,
        "ThetaOne": 65.0,
        "ThetaTwo": 135.0,
        "UnclickThreshold": 0.9700000286102295,
        "UnclickThresholdDrag": 0.9700000286102295,
        "DecayForceOnClick": true,
        "ForceDecayTime": 0.10000000149011612,
        "UseTouchPlaneForce": true,
        "DistPastTouchPlaneMm": 20.0,
        "DragStartDistanceThresholdMm": 30.0,
        "DragDeadzoneShrinkRate": 0.8999999761581421,
        "DragDeadzoneShrinkDistanceThresholdMm": 10.0,
        "DeadzoneMaxSizeIncreaseMm": 20.0,
        "DeadzoneShrinkRate": 0.800000011920929
    },
    "HoverAndHold": {
        "HoverStartTimeS": 0.5,
        "HoverCompleteTimeS": 0.6000000238418579
    },
    "TouchPlane": {
        "TouchPlaneActivationDistanceCM": 5.0,
        "TouchPlaneTrackedPosition": 3
    },
    "VelocitySwipe": {
        "MinScrollVelocity_mmps": 625.0,
        "UpwardsMinVelocityDecrease_mmps": 50.0,
        "DownwardsMinVelocityIncrease_mmps": 50.0,
        "MaxReleaseVelocity_mmps": 200.0,
        "MaxLateralVelocity_mmps": 300.0,
        "MaxOpposingVelocity_mmps": 65.0,
        "ScrollDelayMs": 450.0,
        "MinSwipeLength": 10.0,
        "MaxSwipeWidth": 10.0,
        "SwipeWidthScaling": 0.20000000298023225,
        "AllowBidirectionalScroll": false,
        "AllowHorizontalScroll": true,
        "AllowVerticalScroll": true
    }
}
```

</details>

**Comment by plule** [comment]
> Awesome, that seems to have done the trick.

Cool! I'll do an update so that it works with no modification. Maybe dynamic linking makes sense for windows too

For the setup, I've never seen these json. I usually do it from the control panel, it looks like it's installable on linux too, and once setup there, it is setup for all the application, including leaprs.

![image](https://user-images.githubusercontent.com/630159/215514444-102b0525-a3ea-47f8-bd10-e47984d0637d.png)

Make sure that the hand is detected there, if not change the camera inversion setting and/or the tracking mode, and try launching the example again.

**Comment by MostHated** [comment]
Ah yes, I had set that up prior, but it seem that config was reset to default. I just changed it again and now all is well! This is great. I can't wait to play around with it. I am hoping to use it to issue key presses/run scripts based on hand gestures. This seems like it should allow me to do that. 

```rust
✔ Connected. Service state: (empty)
✔ Got the device LE3100040000004495-3610-693E822028 (3Di)
✔ Tracking mode: ScreenTop
✔ Got a hand
✔ The hand is closed
✔ The hand is opened
✔ Saved image.png
 *  Terminal will be reused by tasks, press any key to close it. 

```

**Comment by plule** [comment]
Looks good, glad to hear this actually work on Linux, I intend to use it as soon as the old leap motion is supported

**Comment by plule** [comment]
@MostHated Just to confirm, the path `/usr/share/doc/ultraleap-hand-tracking-service` is the default installation path? I can take that as a sensible default for Linux?

**Comment by MostHated** [comment]
Yeah, though, that is symlinked to `/usr/lib/ultraleap-hand-tracking-service` upon install

**Comment by plule** [comment]
Just released `v0.1.1` with the fixed `build.rs`
